build_odin.sh

```diff
@@ -52,10 +52,10 @@ config_darwin() {
 		fi
 	fi
 
-	MAX_LLVM_VERSION=("14.999.999")
+	MAX_LLVM_VERSION=("17.999.999")
```

Comment:
This is where things get a little complicated.

We supported LLVM 11, 12, 13, and now 17. We didn't support 14 (even if people got it working) and we definitely did not support 15 and 16. So Just a simple "max version" check is not necessarily a good idea.

Reply:
But maybe this is a non-issue in practice because of the preprocessor check.

Reply:
Yeah I'm curious why the shell script even needs to separately define dependencies other than to error as early as possible.

Reply:
```
$ LLVM_CONFIG=llvm-config-16 ./build_odin.sh 
...
src/llvm_backend.hpp:2:2: error: "LLVM version must be 11, 12, 13, or 17"
    2 | #error "LLVM version must be 11, 12, 13, or 17"
      |  ^
src/llvm_backend.hpp:25:10: fatal error: 'llvm-c/Transforms/AggressiveInstCombine.h' file not found
   25 | #include <llvm-c/Transforms/AggressiveInstCombine.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```
Hoisting the preprocessor check in `llvm_backend.hpp` could make this a little more clear?

Reply:
I think a lot of people are using llvm14 at the moment and making it an error at this point should be avoided

Reply:
I've opted to clean up `build_odin.sh` instead of relying solely on preprocessor macros.